extensions/blocks/podcast-player/view.js

```diff
@@ -44,23 +48,143 @@ window.jetpackPodcastPlayers = {
 	push: initializeBlock,
 };
 
-// Add global handler for clicks.
-window.addEventListener( 'click', function( e ) {
+function handleMediaPlay( e ) {
+	const audioEl = e.detail.target;
+	const parentBlockEl = audioEl.closest( '.wp-block-jetpack-podcast-player' );
+	if ( ! parentBlockEl ) {
+		return;
+	}
+
+	// Clean up any error indication if present
+	const episodeErrorEl = parentBlockEl.querySelector( '.podcast-player__episode-error' );
+	if ( episodeErrorEl ) {
+		parentBlockEl.classList.remove( 'is-error' );
+		episodeErrorEl.remove();
+	}
+
+	parentBlockEl.classList.remove( 'is-paused' );
+	parentBlockEl.classList.add( 'is-playing' );
+}
+
+function handleMediaPause( e ) {
+	const audioEl = e.detail.target;
+	const parentBlockEl = audioEl.closest( '.wp-block-jetpack-podcast-player' );
+
+	parentBlockEl.classList.remove( 'is-playing' );
+	parentBlockEl.classList.add( 'is-paused' );
+}
+
+function handleMediaError( e ) {
+	const audioEl = e.detail.target;
+	const parentBlockEl = audioEl.closest( '.wp-block-jetpack-podcast-player' );
+	const activeEpisodeLinkEl = parentBlockEl.querySelector( '.is-active > a' );
+
+	renderEpisodeError( activeEpisodeLinkEl );
+}
+
+const episodeLinkEls = document.querySelectorAll( '[data-jetpack-podcast-audio]' );
+
+Array.prototype.forEach.call( episodeLinkEls, buildEpisodeLinkClickHandler );
+
+function buildEpisodeLinkClickHandler( episodeLinkEl ) {
+	episodeLinkEl.addEventListener( 'click', handleEpisodeLinkClick );
+	episodeLinkEl.addEventListener( 'keydown', handleEpisodeLinkKeydown );
+}
```

Comment:
This declaration is incompatible with the way players could be initialized asynchronously. Possible solutions:
- delegate these to window so they work regardless of the player registration? (that can definitely be done for the click, not sure about keydown)
- if the global delegation is not an option, we should move adding the listener into the `initializeBlock`, because that is the only time when a presence of episode links is guaranteed

Reply:
Good point. And that's a whole lot fewer listeners added overall.

Reply:
I'm a little worried about having a `window` level `keydown` listener. I'm not sure if that's a valid concern or not though. If it is, should we just add click/keydown listeners to each individual podcast player instance (rather than every episode link)?

Reply:
> I'm a little worried about having a window level keydown listener. 

I don't see any disadvantage of that. It's usually much more preferred way to register events over adding it to every single one.

Reply:
> delegate these to window so they work regardless of the player registration

I remember jQuery had an easy way of doing it 😄:
```js
$(document).on('click', '.podcast-player__episode-link', function() {
  // handle link click
})
```

I guess here we'd just need to find the `closest` episode link element, e.g.:
```js
document.addEventListener('click', function(e) {
  const episodeLinkEl = e.target.closest('.podcast-player__episode-link');
  if (!episodeLinkEl) {
    return;
  }
  // handle link click
})
```

How does that look?

Reply:
It looks almost exactly like what was there before it was rewritten 🙈Let's do it

Reply:
The window event keypress listener was just that it would be potentially listening in on password field entries as well. I don't know if that opens up potential security issues. I'm not a security expert by any means, but that was my only concern. Performance wise it seems better to just have one listener.

Reply:
I'm just not sure if it works the same way for the keyboard interaction (focused element being the `e.target`). If it's not, we might need to use `document.activeElement` specifically in the key handler instead of the target.

Reply:
Let's make sure to `return` as the very first thing in the handler once we detect it's not an event inside the scope we are interested in. That way, we won't affect any other usage, including password elements or other things

Reply:
We may want to use `stopPropagation` as well after we've handled the event.

> I guess here we'd just need to find the closest episode link element, e.g.:

I agree with @marekhrabe. I wonder if we can do something more... "focused" 😉, since `closest` will continue up the entire dom tree.